docker/controller/Dockerfile

```diff
@@ -1,8 +1,32 @@
-FROM centos
+# Start from the latest golang base image
+FROM golang:latest as builder
 
-RUN mkdir -p /opt/tyk-k8s
-COPY files/build/tyk-k8s /opt/tyk-k8s/tyk-k8s
+# Set the Current Working Directory inside the container
+WORKDIR /app
+
+# Copy go mod and sum files
+COPY go.mod go.sum ./
+
+# Download all dependencies. Dependencies will be cached if the go.mod and go.sum files are not changed
+RUN go mod download
+
+# Copy the source from the current directory to the Working Directory inside the container
+COPY . .
+
+# Build the Go app
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o tyk-k8s .
 
+
+######## Start a new stage from scratch #######
+FROM alpine:latest
+
+RUN apk --no-cache add ca-certificates
+
+RUN mkdir -p /opt/tyk-k8s
 WORKDIR /opt/tyk-k8s
 
-CMD ["/opt/goproxy/tyk-k8s", "start"]
\ No newline at end of file
+# Copy the Pre-built binary file from the previous stage
+COPY --from=builder /app/tyk-k8s /opt/tyk-k8s/tyk-k8s
+
+# Command to run the executable
+CMD ["./tyk-k8s", "start"]
```

Comment:
Suggest using an `ENTRYPOINT`:

```.dockerfile
ENTRYPOINT [ "./tyk-k8s" ]
CMD [ "start" ]
```

This way the container could be called with any command to `tyk-k8s` without overriding the call completely, e.g. `docker run tykio/tyk-k8s:latest start` or `docker run tykio/tyk-k8s:latest dosomethingelse`